fedimint-testing/src/lib.rs

```diff
@@ -27,46 +29,48 @@ pub struct TestInputMeta {
     pub keys: Vec<secp256k1_zkp::XOnlyPublicKey>,
 }
 
-impl<M, CC> FakeFed<M, CC>
+impl<Module> FakeFed<Module>
 where
-    M: FederationModule,
-    M::ConsensusItem: Clone,
-    M::TxOutputOutcome: Eq + Debug,
+    Module: FederationModule,
+    Module::ConsensusItem: Clone,
+    Module::TxOutputOutcome: Eq + Debug,
 {
-    pub async fn new<C, F, FF>(
+    pub async fn new<ConfGen, F, FF>(
         members: usize,
         constructor: F,
-        params: &C::Params,
-    ) -> FakeFed<M, C::ClientConfig>
+        params: &ModuleConfigGenParams,
+        conf_gen: &ConfGen,
+    ) -> anyhow::Result<FakeFed<Module>>
     where
-        C: GenerateConfig,
-        F: Fn(C, Database) -> FF, // TODO: put constructor into Module trait
-        FF: Future<Output = M>,
+        ConfGen: FederationModuleConfigGen,
+        F: Fn(ServerModuleConfig, Database) -> FF, // TODO: put constructor into Module trait
```

Comment:
Delete that TODO.

Reply:
I guess for a test that doesn't make much sense because the constructor allows us to mock etc.

Reply:
Yeap.